src/App.tsx

```diff
@@ -1,61 +1,112 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/ToDo';
+
+const posts = todosFromServer.map((todo) => {
+  const user = usersFromServer.find((person) => person.id === todo.userId);
+
+  return { ...todo, user };
+});
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(posts);
+  const [titleField, setTitleField] = useState('');
+  const [userName, setUserName] = useState('');
+
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorName, setErrorName] = useState(false);
+
+  const addPost = (todo: Todo) => {
+    setTodos((prev) => [...prev, todo]);
+  };
+
+  const onChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleField(event.target.value);
+    setErrorTitle(false);
+  };
+
+  const onChangeName = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserName(event.target.value);
+    setErrorName(false);
+  };
+
+  const reset = () => {
+    setUserName('');
+    setTitleField('');
+
+    setErrorName(false);
+    setErrorTitle(false);
+  };
+
+  const sendPost = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const nextId = Math.max(...todos.map((todo) => todo.id)) + 1;
+
+    const objToAdd = {
+      id: +nextId,
+      title: titleField,
+      completed: false,
+      userId: nextId,
+      user: usersFromServer.find((user) => user.name === userName) || null,
+    };
+
+    if (!titleField) {
+      setErrorTitle(true);
```

Comment:
just add return here
```suggestion
      setErrorTitle(true);
      return;
```